packages/dev/core/src/FlowGraph/flowGraphPath.ts

```diff
@@ -0,0 +1,155 @@
+    /**
+     * Whether this path has any template strings in it.
+     */
+    public hasTemplateStrings: boolean = false;
+
+    constructor(path: string) {
+        this._path = path;
+        const { pathParts, templateStrings } = this._getPathPartsAndTemplateStrings(path);
+        this._pathParts = pathParts;
+        this._templateStrings = templateStrings;
+        this.hasTemplateStrings = templateStrings.length > 0;
+    }
+
+    private _getPathPartsAndTemplateStrings(path: string) {
+        const allMatches = path.matchAll(PATH_REGEX);
+
+        const pathParts = [];
+        const templateStrings = [];
+        for (const singleMatch of allMatches) {
+            const [, value] = singleMatch;
+            let valueWithoutBraces = value;
+            let isTemplate = false;
+            if (value.startsWith("{") && value.endsWith("}")) {
+                isTemplate = true;
+                valueWithoutBraces = value.slice(1, value.length - 1);
+                if (templateStrings.indexOf(valueWithoutBraces) === -1) {
+                    templateStrings.push(valueWithoutBraces);
+                }
+            }
+            pathParts.push({
+                value,
+                isTemplate,
+                valueWithoutBraces,
+            });
+        }
+        return { pathParts, templateStrings };
+    }
+
+    /**
+     * Gets the template strings in this path.
+     * @returns an array containing the template strings in this path.
+     */
+    getTemplateStrings(): string[] {
+        return this._templateStrings;
+    }
+
+    setTemplateSubstitution(template: string, value: number) {
+        if (this._templateStrings.indexOf(template) === -1) {
+            throw new Error(`Template string ${template} does not exist in path ${this._path}`);
+        }
+        this._templateSubstitutions[template] = value;
+    }
+
+    private _evaluateTemplates() {
+        for (const pathPart of this._pathParts) {
+            if (pathPart.isTemplate) {
+                const value = this._templateSubstitutions[pathPart.valueWithoutBraces];
+                if (value === undefined) {
+                    throw new Error(`Template string ${pathPart.value} was not substituted`);
+                }
+                pathPart.replacedValue = value.toString();
+            }
+        }
+    }
+
+    private _getFinalPath() {
+        let finalPath = "";
+        for (const pathPart of this._pathParts) {
+            if (pathPart.isTemplate) {
+                finalPath += pathPart.replacedValue;
+            } else {
+                finalPath += pathPart.value;
+            }
+        }
+        return finalPath;
+    }
+
+    /*
+     * Breaks the path into a chain of entities, for example,
+     * /x/y/z would be split into [context._userVariables.x, context._userVariables.x.y, context._userVariables.x.y.z]
+     */
+    private _evaluatePath(context: FlowGraphContext): any {

```

Comment:
I think you could at least return a type like `{ entityChain: any[], splitPath: string[] }` instead of `any`.